enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,470 @@
+
+In 4.19.0 we will address this by pinning to one specific version of the
+Gateway API CRDs that is tested and vetted by us to ensure freedom from "dead
+fields" with the corresponding version of our first-party implementation. We
+fully anticipate that in future iterations we'll need to expand upon this to
+better accomodate third-party solutions.
+
+> **Note**: As we move further into the 4.x.x release cycle we will take
+> feedback and consider continued augmentation to how we manage this and may
+> eventually cover a version range instead of a specific version. We may add
+> additional validation tools that can identify if a field is dead given the
+> implementation being used and reject that. We expect this problem is not a
+> problem unique to us however, and so we will try and pursue this in upstream
+> so that the community has well agreed upon upstream documentation and tools
+> to deal with it, as this will help provide a more consistent user experience
+> for Gateway API users as a whole.
+
+### Drawbacks
+
+> The idea is to find the best form of an argument why this enhancement should
+> _not_ be implemented.
+>
+> What trade-offs (technical/efficiency cost, user experience, flexibility,
+> supportability, etc) must be made in order to implement this? What are the reasons
+> we might not want to undertake this proposal, and how do we overcome them?
+>
+> Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+> aligned with existing user expectations?  Will it be a significant maintenance
+> burden?  Is it likely to be superceded by something else in the near future?
+
+The key tradeoff is that we will start treating Gateway API effectively as if
+it were a core API. This means any perceived value someone might see in being
+able to manage and upgrade the Gateway API resources themselves, independently
+of the platform version, will be lost.
+
+> **Note**: This "downside" can actually be an upside for integrators, as we've
+> heard from projects which have Gateway API integrations (such as OSSM) which
+> appreciate having a consistent and known version of Gateway API on the
+> cluster which they can always rely on being there.
+
+Unfortunately this is not really an avoidable problem, as it is not tenable for
+us to simultaneously say that we support Gateway API as a primary and fully
+supported API surface for ingress traffic, and also have no control over which
+version of the APIs are present, or if they are even present at all (e.g. the
+cluster admin decides to delete them).
+
+To make this situation a bit easier we do anticipate providing some updates
+after the initial release. We want to eventually allow version _ranges_ in
+time after we resolve the "dead fields" problem (see more in the section about
+this above) which we expect to provide significantly more flexibility and take
+care of many concerns that would come from thsi change. We are also tracking and
```

Comment:
```suggestion
care of many concerns that would come from this change. We are also tracking and
```

Reply:
fixed in 01ed78e1ca09cac2fe5617af30985ab43b7cc954 ty!